pkg/terraform/stages/gcp/stages.go

```diff
@@ -42,3 +51,76 @@ func removeFromLoadBalancers(s stages.SplitStage, directory string, terraformDir
 		"failed disabling bootstrap load balancing",
 	)
 }
+
+// extractGCPLBConfig extracts the load balancer information from the terraform outputs, generates the
+// Load Balancer Config file, regenerates the bootstrap ignition, and updates the terraform variables file.
+func extractGCPLBConfig(s stages.SplitStage, directory string, terraformDir string, file *asset.File, tfvarsFile *asset.File) (string, error) {
+	outputs := map[string]interface{}{}
+	err := json.Unmarshal(file.Data, &outputs)
+	if err != nil {
+		return "", err
+	}
+
+	// Extract the Load Balancer ip addresses from the terraform output.
+	apiLBIpRaw, ok := outputs["cluster_public_ip"]
+	if !ok {
+		return "", fmt.Errorf("failed to read External API LB DNS Name from terraform outputs")
+	}
+	apiIntLBIpRaw, ok := outputs["cluster_ip"]
+	if !ok {
+		return "", fmt.Errorf("failed to read Internal API LB DNS Name from terraform outputs")
+	}
+
+	// Create the load balancer configuration file, and add the parsed terraform variables
+	// from above.
+	lbConfigContents, err := lbconfig.CreateLBConfigMap("openshift-lbconfig-custom-dns", apiIntLBIpRaw.(string), apiLBIpRaw.(string))
+	if err != nil {
+		return "", fmt.Errorf("failed to create load balancer config contents: %w", err)
+	}
+	pth := fmt.Sprintf("%s/%s", directory, lbconfig.ConfigName)
+	if err := os.WriteFile(pth, []byte(lbConfigContents), 0o640); err != nil {
+		return "", fmt.Errorf("failed to rewrite %s: %w", lbconfig.ConfigName, err)
+	}
+
+	// Parse the terraform input values. Determine if the install is using a user configured dns solution.
+	tfvarData := map[string]interface{}{}
+	err = json.Unmarshal(tfvarsFile.Data, &tfvarData)
+	if err != nil {
+		return "", err
+	}
+
+	ignitionBootstrap, ok := tfvarData["ignition_bootstrap"]
+	if !ok {
+		return "", fmt.Errorf("failed to read ignition bootstrap from tfvars")
+	}
+
+	ignData := igntypes.Config{}
+	err = json.Unmarshal([]byte(ignitionBootstrap.(string)), &ignData)
+	if err != nil {
+		return "", err
+	}
+	path := "/opt/openshift/manifests/openshift-lbconfig-custom-dns.yaml"
```

Comment:
Can we name the manifest something else? Name should indicate contents rather than what is is used for (that could change in the future.) 
How about openshift-lbconfig.yaml?